### PR TITLE
Cookies creation spike

### PR DESCRIPTION
**Pull request by JamesC-MHCLG**
Create cookie banner and page components.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,8 +3,6 @@
 .build_cache
 ### VisualStudioCode template
 .vscode/*
-!.vscode/settings.json
```
Comment (Jordan-Hall) [inline .gitignore:6]: Is this change needed?

File: src/demo/App.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
 /* eslint no-magic-numbers: 0 */
 import React, { useState } from 'react';
 
-import { Accordion, CheckboxList, ComponentTemplate, Dropdown, ExpandableMenuItem, Radios, Tabs } from '../lib';
+import { BrowserRouter as Router, Routes, Route, Link } from 'react-router-dom';
+
+
+import { Accordion, CheckboxList, ComponentTemplate, Dropdown, ExpandableMenuItem } from '../lib';
+
+import Tabs from '../lib/fragments/Tabs.react';
```
Comment (Jordan-Hall) [inline src/demo/App.js:9]: Should all import from ../lib

File: src/lib/LazyLoader.js [hunk 3]
```diff
@@ -8,9 +8,25 @@ import ComponentTemplate from './fragments/ComponentTemplate.react'
...
+	Accordion,
+	AdditionalDetails,
+	AutoComplete,
+	ChangeLogBanner,
+	CheckboxList,
+	ComboBox,
+	ComponentTemplate,
+	Dropdown,
+	ExpandableMenuItem,
+	Tabs,
```
Comment (Jordan-Hall) [inline src/lib/LazyLoader.js:25]: Linting ?

File: src/lib/components/cookies/utils/Cookie.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        // GovUK tracker
+        gtag('config', tag, {
+            cookie_domain: 'none'
+        });
+
+
+        console.info("GA successfully set up.", gtag);
+
+    } catch (error) {
+        console.warn("Cookies accepted, but tracking is blocked by the browser.")
```
Comment (Jordan-Hall) [inline src/lib/components/cookies/utils/Cookie.js:41]: Are these requires? It's not normally ideal for console logging for production

File: src/lib/fragments/CookieBanner.react.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import React, { useState, useEffect } from 'react';
+
+import { setCookies, deleteCookies, handleCookieAccept } from '../components/cookies/utils/Cookie';
+
+import Cookies from "js-cookie";
+
+
+const CookieBanner = ({ ...props }) => {
+    console.log("start of cookiebanner")
+    console.trace()
```
Comment (kateriley1) [inline src/lib/fragments/CookieBanner.react.js:11]: needs removing

File: src/lib/fragments/CookieBanner.react.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                </div>
+
+                {/* .govuk-cookie-banner .govuk-button-group {
+  display: block;
+} needs removing from pkg css */}
+                <div className="govuk-button-group">
+                    <button className="govuk-button" type="submit"
+                        data-module="track-click" data-accept-cookies="true"
+                        data-track-category="cookieBanner"
+                        style={{ maxWidth: '50%' }}
```
Comment (Jordan-Hall) [inline src/lib/fragments/CookieBanner.react.js:126]: Can this not move to css?

File: src/lib/fragments/CookiePages/Cookies.styles.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import styled from 'styled-components';
+import{ ComponentType } from 'react';
+
+
+export const Article = (() =>
+    styled.article`
```
Comment (Jordan-Hall) [inline src/lib/fragments/CookiePages/Cookies.styles.js:7]: Css

File: src/lib/fragments/CookiePages/Cookies.styles.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+        max-width: 50em;
+    `
+)();
+
+
+export const Heading = ({ className = '', ...props }) => (
```
Comment (Jordan-Hall) [inline src/lib/fragments/CookiePages/Cookies.styles.js:13]: Can this be in a typography tsx and then it's reusable to gov standards. It good for DSL to have dedicated typography file ans components

File: src/lib/fragments/CookiePages/CookiesPage.js [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+        </p>
+      </div>
+    </div>
+  );
+};
+
+const CookiesPage = ({ ...props }) => {
+    const [cookieState, setCookieState] = useState(null);
+    const [updateState, setUpdateState] = useState(false);
+    const { tag } = props;
```
Comment (Jordan-Hall) [inline src/lib/fragments/CookiePages/CookiesPage.js:33]: Would it be worth using context for this and the dispatch.  Rather than reimplmenting this. Or just custom hook with its own useEffect in it

File: webpack.config.js [hunk 10]
```diff
@@ -94,7 +94,7 @@ module.exports = (env, argv) => {
                 })
             ],
             splitChunks: {
-                name: true,
+                name: false,
```
Comment (Jordan-Hall) [inline webpack.config.js:97]: Why this is a bad move normally leading to bloat

**Review by Jordan-Hall — COMMENTED** [review]
Small suggestions. Looking some duplication too

File: src/lib/LazyLoader.js (hunk 3 above)
Comment (kateriley1) [reply src/lib/LazyLoader.js:25]: we thought it looked easier to read & understand vertically - it shouldn't affect the functionality?

File: src/lib/components/cookies/utils/Cookie.js (hunk 4 above)
Comment (JamesC-MHCLG) [reply src/lib/components/cookies/utils/Cookie.js:41]: We were intending to leave these in until we've tested the analytics in dev.

File: .gitignore (hunk 1 above)
Comment (kateriley1) [reply .gitignore:6]: vs.code/settings.json is personal to each users workspace.

File: src/lib/fragments/CookiePages/Cookies.styles.js (hunk 8 above)
Comment (JamesC-MHCLG) [reply src/lib/fragments/CookiePages/Cookies.styles.js]: We've created a ticket to build a typography file.

**Review by kateriley1 — APPROVED** [review]
